pydis_site/apps/main/views/home.py

```diff
@@ -0,0 +1,106 @@
+import requests
+from django.conf import settings
+from django.shortcuts import render
+from django.utils import timezone
+from django.views import View
+
+from pydis_site.apps.main.models import RepoData
+
+GITHUB_API = "https://api.github.com/users/python-discord/repos"
+
+
+class Home(View):
+
+    def _get_api_data(self):
+        """Call the GitHub API and get information about our repos."""
+        repo_dict = {repo_name: {} for repo_name in settings.HOMEPAGE_REPOS}
+
+        # Fetch the data from the GitHub API
+        api_data = requests.get(GITHUB_API)
+        api_data = api_data.json()
+
+        # Process the API data into our dict
+        print(f"repo_dict = {repo_dict}")
+        for repo in api_data:
+            full_name = repo["full_name"]
+
+            if full_name in settings.HOMEPAGE_REPOS:
+                repo_dict[full_name] = {
+                    "full_name": repo["full_name"],
+                    "description": repo["description"],
+                    "language": repo["language"],
+                    "forks_count": repo["forks_count"],
+                    "stargazers_count": repo["stargazers_count"],
+                }
+        print(f"repo_dict after processing = {repo_dict}")
+        return repo_dict
+
+    def _get_repo_data(self):
+        """Build a list of RepoData objects that we can use to populate the front page."""
+
+        # Try to get site data from the cache
+        try:
+            repo_data = RepoData.objects.get(repo_name="python-discord/site")
+
+            # If the data is older than 2 minutes, we should refresh it. THIS PROBABLY ALWAYS FAILS?
+            if (timezone.now() - repo_data.last_updated).seconds > 120:
+
+                diff = (timezone.now() - repo_data.last_updated).seconds
+                print(f"okay baby, it's old! the seconds difference comes to: {diff}")
+
+                # Get new data from API
+                api_data_container = self._get_api_data()
+                repo_data_container = []
```

Comment:
I feel like this "update if not updated recently" is something we'll reuse a bunch and maybe having this as a separate decorator would be useful

Reply:
I don't know, I think that sounds a bit over-engineered until we actually have that problem. Or maybe I'm just too lazy to do it. If we ever need that pattern again, we should probably do that, though.